hlx_statics/blocks/redoclyapiblock/redoclyapiblock.js

```diff
@@ -0,0 +1,133 @@
+
+const DEFAULT_OPTIONS = {
+  src: 'https://raw.githubusercontent.com/AdobeDocs/adp-devsite-github-actions-test/refs/heads/main/static/openapi.yaml',
+  width: '500px',
+  typography: 'fontFamily: `adobe-clean, "Source Sans Pro", -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Ubuntu, "Trebuchet MS", "Lucida Grande", sans-serif`',
+  codeBlock: 'tokens: { punctuation: { color: "white" }}',
+  disableSidebar: false,
+  disableSearch: false,
+  hideTryItPanel: false,
+  scrollYOffset: 0,
+  sortOperationsAlphabetically: false,
+  sortTagsAlphabetically: false,
+  jsonSampleExpandLevel: 2,
+  generateCodeSamples: 'languages: [], skipOptionalParameters: false,',
+  requestInterceptor: '',
+};
+
+const ATTRIBUTE_PREFIX = 'data-';
+
+function parseOptions(block) {
+  // start with default options
+  const options = Object.assign({}, DEFAULT_OPTIONS);
+
+  const blockChildrenToRemove = [];
+
+  // overwrite with options from user
+  block.querySelectorAll('div > div > pre > code').forEach(code => {
+    if(code.textContent.startsWith(ATTRIBUTE_PREFIX)){
+      let [name, value] = code.textContent.split("=");
+      name = name.replace(ATTRIBUTE_PREFIX, '');
+      value = value.trim();
+
+      // JSX shorthand for passing true
+      if(value === 'undefined') {
+        value = true;
+      }
+      
+      options[name] = value;
+
+      // remove attribute row since already been parsed
+      blockChildrenToRemove.push(code.parentElement.parentElement.parentElement);
+    }
+  });
+
+  block.querySelectorAll('div > div > strong').forEach(strong => {
+    if(strong.textContent.startsWith(ATTRIBUTE_PREFIX)) {
+       // remove comma-separated attributes row as this is redundant with the ones already parsed above
+      blockChildrenToRemove.push(strong.parentElement.parentElement);
+    } 
+  })
+
+  // clear attribute rows
+  blockChildrenToRemove.forEach(child => block.removeChild(child));
+
+  return options;
+}
+
+export default function decorate(block) {
+  const { 
+      src,
+      width,
+      typography,
+      codeBlock,
+      disableSidebar,
+      disableSearch,
+      hideTryItPanel,
+      scrollYOffset,
+      sortOperationsAlphabetically,
+      sortTagsAlphabetically,
+      jsonSampleExpandLevel,
+      generateCodeSamples,
+      requestInterceptor,
+    } = parseOptions(block);
+  
+  const absoluteSrc = getResourceUrl(src);
+
+  // https://redocly.com/docs/api-reference-docs/guides/on-premise-html-element#steps
+  const redocly_container = createTag('div', {id: 'redocly_container'});
+  block.appendChild(redocly_container);
+
+  const redocly = createTag('script', {
+      src: 'https://cdn.redoc.ly/reference-docs/latest/redocly-reference-docs.min.js', 
+      async: true
+    });  
+
+  const console = createTag('script', {
```

Comment:
what is the difference between redocly and console?

Reply:
Thanks for catching that! Fixed in a7199215ea6678898fda3d9dfc1577fc710a5b43